BZ2025482: Required Azure roles (#39852)
openshift/openshift-docs: modules/installation-azure-permissions.adoc

[id="installation-azure-permissions_{context}"]
= Required Azure roles

> Your Microsoft Azure, which will be used to create the service principal used by the installer and cluster, must have the following roles for the subscription that you use:

Reviewer: I was a bit confused with this. Breaking up the sentence may help make it clearer:

"{product-title} needs a service principal so it can manage Microsoft Azure resources. Before you can create a service principal, your Azure account subscription must have the following roles:"

Author: I agree. Above sentence is clearer

Author: Addressed

> {product-title} needs a service principal so it can manage Microsoft Azure resources. Before you can create a service principal, your Azure account subscription must have the following roles:

Reviewer: @patrickdillon We've change the wording for clarity. Is this still okay?

Reviewer 2: Yes LGTM.

[id="installation-azure-permissions_{context}"]
= Required Azure roles

> You can use your Microsoft Azure account for `az login` and browsing in the web console. By using your Microsoft Azure account you can run the installer and create the cluster. Your Microsoft Azure account must have the following roles for the subscription

Reviewer 3: > By using your Microsoft Azure account you can run the installer and create the cluster

I think the statement is wrong, The Microsoft Azure account has the correct permissions to create the service principal, the created service principal account will be used to run the installer and create the cluster.

Author: Changed the sentence. Managed to maintain the present tense. Let me know if any change is required!

> You can use your Microsoft Azure account for `az login` and browsing in the web console. The Microsoft Azure account has the correct permissions to create the service principal account. The created service principal account is used to run the installer and create the cluster. Your Microsoft Azure account must have the following roles for the subscription

Reviewer 2: ```suggestion
Your Microsoft Azure, which will be used to create the service principal used by the installer and cluster, must have the following roles for the subscription
```

Author: addressed. Thanks for the review

> You can use your Microsoft Azure account for `az login` and browsing in the web console. The Microsoft Azure account has the correct permissions to create the service principal account. The Microsoft Azure account uses the created service principal account to run the installer and create the cluster. Your Microsoft Azure account must have the following roles for the subscription

Reviewer 3: How about changing "The Microsoft Azure account uses the created service principal account to run the installer and create the cluster" to "The created service principal account is used to run the installer and create the cluster" ?

Author: acked

that you use:

* `User Access Administrator`

> * `Owner`

Reviewer 2: How did we determine that the Owner role is required? Is the User Access Administrator role alone not sufficient?  We should only add the bare minimum amount of required roles.

Reviewer 3: >  Is the User Access Administrator role alone not sufficient?
Right. QE tried that. https://bugzilla.redhat.com/show_bug.cgi?id=2025482#c9

Reviewer 2: Ok this makes sense. Thank you.

> that you use:

Reviewer 3: To help resolve customer confusion, maybe we can add some more description about the Azure USER account purpose. 
Form Patrick's Bugzilla comments:
```
This is the user account for `az login` or when browsing the web portal. The purpose of this section is to make sure that the user has the correct permissions to create the service principal which will then be used to run the installer and the cluster.
```

Author: acked

that you use:

* `User Access Administrator`

> * `Contributor`

Reviewer 3: If the user already has the `Owner` role, I guess it does not need `Contributor` any more.

Author: ack
